### DISCUSSION TITLE
How to have always identifier when serialize ?

### DISCUSSION THREAD

**Discussion by JoshuaBaubry** [api-platform/core#5146] [Q&A]
Hello,

I have a Profile entity with childs and parents :
```
class Profile
{
 /**
     * Parents profiles of this profile
     * @var Profile[]
     *
     * @ORM\ManyToMany(targetEntity="App\Entity\Profile",inversedBy="childrens")
     * @ORM\JoinTable(name="profiles_profiles",joinColumns={@ORM\JoinColumn(name="profile_id")},
     *      inverseJoinColumns={@ORM\JoinColumn(name="dependency_id")})
     * @ORM\OrderBy({"idText" = "ASC"})
     * @MaxDepth(1)
     * @Groups({"read_profile","write_profile"})
     */
    private $parents;

    /**
     * Childrens profiles of this profiles
     * @var Profile[]
     *
     * @ORM\ManyToMany(targetEntity="App\Entity\Profile",mappedBy="parents")
     * @ORM\OrderBy({"idText" = "ASC"})
     * @Groups({"read_profile","write_profile"})
     * @MaxDepth(1)
     */
    private $childrens;
}
```

And when I seriliaze profile I have : 
```
{
  "id": 298,
  "parents": [
      {
      "id": 337,
      "childrens": [...],
          ...
      }
    ],
     ...
}
```

But I don't need any serilization for parents and I just need an IRI : 
```
{
  "id": 298,
  "parents": [
      "/api/profiles/337"
    ],
     ...
}
```
I finded #1696 which advises to use MaxDepth, but even with a MaxDepth of 1 the parent the parent serializes once. 

Maybe I did not understand something, can you help me?

**Comment by soyuka** [comment]
With serialization groups, if the children don't have any I think that it'll output that result. See [this test](https://github.com/api-platform/core/blob/master/features/main/relation.feature#L133) and their related entities in our fixtures!

**Comment by JoshuaBaubry** [comment]
Thanks for your reply. I know what you are talking about (and I use this for my other entities in relations) but here parent entity and children's are the same type, so they have the same serialization group.

**Comment by soyuka** [comment]
Oh sorry I missed that. There must be a way to do this, a custom normalizer for example or maybe there's something even simpler using the serialization context, not sure rn ^^.

**Comment by teohhanhui** [comment]
It is a feature that needs to be introduced in the Symfony Serializer. I came up with a similar example to illustrate the problem:

```json
{
  "code": "ABC",
  "children": [
    {
      "code": "DEF",
      "children": [
        {
          "code": "JKL",
          "children": []
        }
      ]
    },
    {
      "code": "XYZ",
      "children": []
    }
  ],
  "parent": {
    "code": "QRS",
    "parent": null
  }
}
```

I want to keep traversing down `children`, and keep traversing up `parent`, but not both at the same time.

It's not currently possible to have such control over the serialization output. I think we need some kind of property path / "scope" support.
